vllm/model_executor/layers/fused_moe/fused_marlin_moe.py

```diff
@@ -178,8 +213,47 @@ def fused_marlin_moe(
                             device="cuda",
                             requires_grad=False)
 
-    scalar_type = (scalar_types.uint4b8
-                   if num_bits == 4 else scalar_types.uint8b128)
+    if has_zero_point:
+        assert w1_zeros is not None and w1_zeros.nelement() > 0
+        assert w2_zeros is not None and w2_zeros.nelement() > 0
+
+    if w1_zeros is None:
+        w1_zeros = torch.empty((0),
+                               dtype=hidden_states.dtype,
+                               device=hidden_states.device,
+                               requires_grad=False)
+    if w2_zeros is None:
+        w2_zeros = torch.empty((0),
+                               dtype=hidden_states.dtype,
+                               device=hidden_states.device,
+                               requires_grad=False)
+
+    if g_idx1 is None:
+        g_idx1 = torch.empty((0, 0),
+                             dtype=torch.int32,
+                             device=hidden_states.device,
+                             requires_grad=False)
+
+    if g_idx2 is None:
+        g_idx2 = torch.empty((0, 0),
+                             dtype=torch.int32,
+                             device=hidden_states.device,
+                             requires_grad=False)
```

Comment:
Is it possible to have g_idx1 defined and not g_idx2? Is this true for each of these? It seems like no based on the zero point implementation

Reply:
We probably never have such a scenario. I've added a bunch of assertions for this in the beginning of `fused_marlin_moe()` function